pkg/api/query/v1.go

```diff
@@ -224,6 +227,19 @@ func (qapi *QueryAPI) parsePartialResponseParam(r *http.Request, defaultEnablePa
 	return defaultEnablePartialResponse, nil
 }
 
+func (qapi *QueryAPI) parseStep(r *http.Request, defaultRangeQueryStep time.Duration) (step time.Duration, _ *api.ApiError) {
+	// Overwrite the cli flag when provided as a query parameter.
+	if val := r.FormValue("step"); val != "" {
+		var err error
+		defaultRangeQueryStep, err = parseDuration(val)
+		if err != nil {
+			return 0, &api.ApiError{Typ: api.ErrorBadData, Err: errors.Wrapf(err, "'%s' parameter", defaultRangeQueryStep)}
+		}
+	}
+
+	return defaultRangeQueryStep, nil
```

Comment:
Wonder would it makes sense to mimic UI here and do `min(range/250, default step)`, just for consistency with UI? I would love to have same default behaviours 🤗

Reply:
Sorry missed it 😅 . I'll correct it.